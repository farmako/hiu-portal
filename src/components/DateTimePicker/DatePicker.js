import React from 'react';
import * as PropTypes from 'prop-types';
import { KeyboardDatePicker } from '@material-ui/pickers';

const DatePicker = ({
  disableFuture,
  minDate,
  selectedDate,
  handleDateChange,
}) => (
  <KeyboardDatePicker
    autoOk="true"
    variant="inline"
    margin="normal"
    disableFuture={disableFuture}
    id="date-picker-dialog"
    minDate={minDate}
    format="dd/MM/yyyy"
    value={selectedDate}
    onChange={handleDateChange}
    KeyboardButtonProps={{
      'aria-label': 'change date',
    }}
  />
);

DatePicker.propTypes = {
  minDate: PropTypes.instanceOf(Date),
  disableFuture: PropTypes.bool,
  selectedDate: PropTypes.instanceOf(Date).isRequired,
  handleDateChange: PropTypes.func.isRequired,
};

DatePicker.defaultProps = {
  minDate: new Date(1900, 1, 1),
  disableFuture: false,
};

export default DatePicker;
