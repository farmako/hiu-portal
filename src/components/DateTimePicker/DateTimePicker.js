import React from 'react';
import * as PropTypes from 'prop-types';
import { KeyboardDateTimePicker } from '@material-ui/pickers';

const DateTimePicker = ({
  disableFuture,
  minDate,
  maxDate,
  selectedDate,
  handleDateChange,
  disablePast,
}) => (
  <KeyboardDateTimePicker
    autoOk="true"
    variant="inline"
    margin="normal"
    disableFuture={disableFuture}
    id="date-picker-dialog"
    format="dd/MM/yyyy HH:mm"
    minDate={minDate}
    maxDate={maxDate}
    value={selectedDate}
    onChange={handleDateChange}
    disablePast={disablePast}
    KeyboardButtonProps={{
      'aria-label': 'change date',
    }}
  />
);

DateTimePicker.propTypes = {
  minDate: PropTypes.instanceOf(Date),
  maxDate: PropTypes.instanceOf(Date),
  disableFuture: PropTypes.bool,
  disablePast: PropTypes.bool,
  selectedDate: PropTypes.instanceOf(Date).isRequired,
  handleDateChange: PropTypes.func.isRequired,
};

DateTimePicker.defaultProps = {
  minDate: new Date(1900, 1, 1),
  maxDate: new Date(2100, 1, 1),
  disableFuture: false,
  disablePast: false,
};

export default DateTimePicker;
