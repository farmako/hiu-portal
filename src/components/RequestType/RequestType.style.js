import styled from 'styled-components';

const checkboxStyles = styled.div`
  .MuiFormGroup-root {
    flex-direction: row;
  }
  .MuiFormControlLabel-root {
    width:40%;
  }
`;

export default checkboxStyles;
