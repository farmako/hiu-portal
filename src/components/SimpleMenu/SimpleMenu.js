import React from 'react';
import * as PropTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import SimpleMenuStyles from './SimpleMenu.style';

const SimpleMenu = ({ menuItems, selectedValue, handleChange }) => (
  <SimpleMenuStyles>
    <Select
      id="select-menu"
      className="select-menu"
      value={selectedValue}
      onChange={handleChange}
    >
      {menuItems.map((type) => (
        <MenuItem
          id="menu-item"
          key={type.value}
          value={type.value}
          onClick={handleChange}
        >
          {type.label}
        </MenuItem>
      ))}
    </Select>
  </SimpleMenuStyles>
);

SimpleMenu.propTypes = {
  menuItems: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })),
  selectedValue: PropTypes.string,
  handleChange: PropTypes.func,
};

SimpleMenu.defaultProps = {
  menuItems: [],
  selectedValue: '',
  handleChange: {},
};

export default SimpleMenu;
