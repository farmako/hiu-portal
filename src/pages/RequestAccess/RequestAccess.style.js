import styled from 'styled-components';

const RequestAccessStyles = styled.div`
  position: relative;
  height: 100%;

  .form-container {
    height: calc(100% - 110px);
    padding-right: 10px;
    overflow-y: auto;
  }

  h2 {
    color: #000000;
    font-size: 24px;
    line-height: 24px;
    font-weight: 300;
    padding: 20px 0;
    margin-bottom: 25px;
    border-bottom: 1px solid #000000;
    font-family: "Roboto", sans-serif;
  }
  .text-field-label {
    color: #000;
  }
  .label {
    display: block;
    color: #000;
    font-size: 16px;
    font-weight: 300;
    font-family: "Roboto", sans-serif;
    margin-bottom: 12px;
  }
  .MuiGrid-container {
    padding: 10px 0;
  }
  .search-bar {
    vertical-align: center;
  }
  .auto-complete {
    color: #009688;
    .text-field {
      color: #009688;
    }
  }
  .create-consent-button {
    margin-top: 25px;
  }
  .error {
    display: block;
    padding: 20px 0 0px;
    color: #f44336;
    font-size: 16px;
    font-weight: 300;
    font-family: "Roboto", sans-serif;
  }
  .loader-container {
    position: absolute;
    z-index: 99;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    background: #fff;
    text-align: center;
  }
  .loader {
    margin-top: 40vh;
  }
  .request-type-label{
    margin-top:20px;
  }
`;

export default RequestAccessStyles;
