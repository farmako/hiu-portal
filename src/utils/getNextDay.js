function getNextDay() {
  const currentDate = new Date();
  currentDate.setDate(currentDate.getDate() + 1);
  return currentDate;
}

export default getNextDay;
