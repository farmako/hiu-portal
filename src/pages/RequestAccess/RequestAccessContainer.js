import { connect } from 'react-redux';
import RequestAccess from './RequestAccess';
import { onCreateConsent } from '../../redux/actions/createConsentActions';
import getNestedObject from '../../utils/getNestedObject';

const mapStateToProps = (state) => ({
  patientId: getNestedObject(state, 'patientDetail.patientData.patient.id'),
  loading: state.createConsent.loading,
  error: state.createConsent.error,
  purposesOfUse: state.configValueSets.purposesOfUse,
  hiTypes: state.configValueSets.hiTypes,
});

const mapDispatchToProps = {
  onCreateConsent,
};

export default connect(mapStateToProps, mapDispatchToProps)(RequestAccess);
