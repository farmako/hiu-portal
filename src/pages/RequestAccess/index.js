import RequestAccessContainer from './RequestAccessContainer';

export default RequestAccessContainer;
