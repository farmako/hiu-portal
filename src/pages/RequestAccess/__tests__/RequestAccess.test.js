import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '@material-ui/core';
import RequestAccess from '../RequestAccess';
import DatePicker from '../../../components/DateTimePicker/DatePicker';
import DateTimePicker from '../../../components/DateTimePicker/DateTimePicker';
import SimpleMenu from '../../../components/SimpleMenu/SimpleMenu';
import RequestType from '../../../components/RequestType/RequestType';
import getNextDay from '../../../utils/getNextDay';

jest.mock('./../../../utils/getNextDay');

describe('Request Access', () => {
  const mockPurposeOfUse = [
    {
      code: 'ReferralService',
      display: 'Referral Service',
    },
    {
      code: 'BTG',
      display: 'Break the Glass',
    },
  ];
  const mockHiTypes = [
    {
      code: 'Condition',
      display: 'Condition',
    },
  ];

  const currentDate = new Date('2019-05-14T11:01:58.135Z');
  getNextDay.mockImplementation(() => new Date('2019-05-15T11:01:58.135Z'));
  global.Date = class extends Date {
    constructor(date) {
      if (date) {
        return super(date);
      }
      return currentDate;
    }
  };
  const onCreateConsent = jest.fn();

  const wrapper = shallow(
    <RequestAccess
      onCreateConsent={onCreateConsent}
      purposesOfUse={mockPurposeOfUse}
      hiTypes={mockHiTypes}
      patientId="sample@ncg"
      loading={false}
      error={false}
    />,
  );

  it('Should render properly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(DatePicker).length).toEqual(2);
    expect(wrapper.find(DateTimePicker).length).toEqual(1);
  });

  it('Purpose of request, should have referral services selected by default', () => {
    expect(wrapper.find(SimpleMenu).props().selectedValue).toEqual(
      'ReferralService'
    );
  });

  it('From: date component, should have current date when no date is selected ', () => {
    expect(
      wrapper.find(DatePicker).at(0).props().selectedDate.toISOString()
    ).toEqual('2019-05-14T11:01:58.135Z');
  });

  it('To: date component, should have current date when no date is selected ', () => {
    expect(
      wrapper.find(DatePicker).at(1).props().selectedDate.toISOString()
    ).toEqual('2019-05-14T11:01:58.135Z');
  });


  it('Expiry: date component, should have next day as current date when no date is selected ', () => {
    const ActualExpiryDate = wrapper
      .find(DateTimePicker)
      .props()
      .selectedDate.toISOString();
    const expectedExpiryDate = '2019-05-15T11:01:58.135Z';
    expect(ActualExpiryDate).toEqual(expectedExpiryDate);
  });

  it('Error message, should not be shown when no error occurs while creating consent', () => {
    const errorMessage = 'Error occured while creating consent.';
    expect(wrapper.contains(errorMessage)).toBe(false);
  });

  it('Error message, should be shown when error occurs while creating consent', () => {
    wrapper.setProps({ error: true });
    const errorMessage = 'Error occured while creating consent.';
    expect(wrapper.contains(errorMessage)).toBe(true);
  });

  it('Request Consent: Button, should be disabled if none of the HI types are selected', () => {
    expect(wrapper.find(Button).props().disabled).toBe(true);
  });

  it('Request Consent: Button, should be enabled when at least one HI type is selected', () => {
    const handleHITypeChange = wrapper.find(RequestType).props().handleChange;
    const event = { target: { checked: true } };
    handleHITypeChange('PatientHistory')(event);

    expect(wrapper.find(Button).props().disabled).toBe(false);
  });

  it('Request Consent: Button, call onCreateConsent on clicking if there are no errors', () => {
    wrapper.find(Button).props().onClick();
    expect(onCreateConsent).toHaveBeenCalled();
  });

  it('Request Consent: Button, should show error if patientID is empty on click', () => {
    wrapper.setProps({ patientId: '' });
    wrapper.find(Button).props().onClick();

    const errorMessage = 'Please enter a patient identifier';
    expect(wrapper.contains(errorMessage)).toBe(true);
  });

  it('Loading: Circular Progress, should appear when request is in loading state', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.find('#loader').exists()).toBe(true);
  });
});
