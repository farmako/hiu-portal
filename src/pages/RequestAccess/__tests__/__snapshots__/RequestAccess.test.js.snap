// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Access Should render properly 1`] = `
<styled.div>
  <h2>
    Consent request form
  </h2>
  <div
    className="form-container"
  >
    <form>
      <span
        className="label"
      >
        All the fields are mandatory.
      </span>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <WithStyles(ForwardRef(InputLabel))
            className="text-field-label"
          >
            Patient Identifier
          </WithStyles(ForwardRef(InputLabel))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={5}
          sm={6}
          xs={12}
        >
          <Connect(SearchPatient) />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <WithStyles(ForwardRef(InputLabel))
            className="text-field-label"
          >
            Purpose of request
          </WithStyles(ForwardRef(InputLabel))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={5}
          sm={6}
          xs={12}
        >
          <SimpleMenu
            handleChange={[Function]}
            menuItems={
              Array [
                Object {
                  "label": "Referral Service",
                  "value": "ReferralService",
                },
                Object {
                  "label": "Break the Glass",
                  "value": "BTG",
                },
              ]
            }
            selectedValue="ReferralService"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <WithStyles(ForwardRef(InputLabel))
            className="text-field-label"
          >
            Health info from
          </WithStyles(ForwardRef(InputLabel))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={5}
          sm={6}
          xs={12}
        >
          <DatePicker
            disableFuture={true}
            handleDateChange={[Function]}
            minDate={1900-01-31T18:38:50.000Z}
            selectedDate={2019-05-14T11:01:58.135Z}
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <WithStyles(ForwardRef(InputLabel))
            className="text-field-label"
          >
            Health info to
          </WithStyles(ForwardRef(InputLabel))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={5}
          sm={6}
          xs={12}
        >
          <DatePicker
            disableFuture={true}
            handleDateChange={[Function]}
            minDate={2019-05-14T11:01:58.135Z}
            selectedDate={2019-05-14T11:01:58.135Z}
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          className="request-type-label"
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <WithStyles(ForwardRef(InputLabel))
            className="text-field-label"
          >
            Health info type
          </WithStyles(ForwardRef(InputLabel))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          sm={8}
          xs={12}
        >
          <RequestType
            handleChange={[Function]}
            requestTypes={
              Array [
                Object {
                  "label": "Condition",
                  "value": "Condition",
                },
              ]
            }
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={3}
          sm={4}
          xs={12}
        >
          <WithStyles(ForwardRef(InputLabel))
            className="text-field-label"
          >
            Consent Expiry
          </WithStyles(ForwardRef(InputLabel))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md={5}
          sm={6}
          xs={12}
        >
          <DateTimePicker
            disableFuture={false}
            disablePast={true}
            handleDateChange={[Function]}
            maxDate={2100-01-31T18:30:00.000Z}
            minDate={1900-01-31T18:38:50.000Z}
            selectedDate={2019-05-15T11:01:58.135Z}
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="center"
        spacing={2}
      >
        <WithStyles(ForwardRef(Button))
          className="create-consent-button"
          color="primary"
          disabled={true}
          onClick={[Function]}
          variant="contained"
        >
          Request Consent
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </div>
</styled.div>
`;
